### DISCUSSION TITLE
DockerOperator printing the Logs 3x upon task failure

### DISCUSSION THREAD

**Discussion by ujwaltrivedi** [apache/airflow#29431] [Q&A] [answered]
Hello,

I wanted to check if anyone here has faced issues running DocketOperator, printing the logs 3x upon task failure. I suspect this is due to the steaming logs

Stream logging enabled here
https://github.com/apache/airflow/blob/main/airflow/providers/docker/operators/docker.py#L365

Stream logs getting logged here
https://github.com/apache/airflow/blob/main/airflow/providers/docker/operators/docker.py#L373

This is where when the tasks fails, it relogs everything in the logs again, which is causing the failed jobs to log the same thing multiple times.
https://github.com/apache/airflow/blob/main/airflow/providers/docker/operators/docker.py#L382

I wanted to see if this is an ongoing which anyone here has experienced and there are any other solutions before I go an create a feature/bug fix request.

Thanks
Ujwal

**Comment by hussein-awala** [comment, marked as answer]
The idea from `raise AirflowException(f"Docker container failed: {repr(result)} lines {joined_log_lines}")` was to raise an exception with the same error message raised in the Docker container. But it's hard to detect the beginning of the error message, so we raise it with the full log which may be a bad idea when the log is too long.

Feel free to open a PR if you have a better idea.
